### PR TITLE
New CreateMap API implementing bpf_map_create, Type/name accessors

### PR DESCRIPTION
**Pull request by grantseltzer**

### CODE REVIEW

**Review by kakkoyun — COMMENTED** [review]
Looks awesome! Thanks for handling it, and examples are very self-explanatory. I think I'm gonna find some spare time next week to integrate this to Parca. I'll let you know how it goes.

I have just added a comment about tests in BPF program.

File: selftest/create-map/main.bpf.c [hunk 1]
```diff
@@ -0,0 +1,11 @@
+//+build ignore
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+
+SEC("kprobe/sys_execve")
+int kprobe__sys_execve(struct pt_regs *ctx)
```
[the remark this answers is not in the document]
Comment (grantseltzer) [reply selftest/create-map/main.bpf.c:6]: Sure thing!

[the remark this answers is not in the document]
Comment (kakkoyun) [reply selftest/create-map/main.bpf.c:6]: :100: Thanks!

**Review by kakkoyun — APPROVED** [review]

**Comment by grantseltzer** [comment]
@yanivagman can you take a look at this when you have spare time

File: libbpfgo.go [hunk 2]
```diff
@@ -341,6 +470,30 @@ func (m *Module) GetMap(mapName string) (*BPFMap, error) {
 	}, nil
 }
 
+func (b *BPFMap) Name() string {
+	cs := C.bpf_map__name(b.bpfMap)
```
Comment (yanivagman) [inline libbpfgo.go:474]: Quoting from CreateMap above:

> // and therefore do not have access to libbpf high level bpf_map__* APIS

So, this won't work for maps created with CreateMap, right?

File: libbpfgo.go [hunk 3]
```diff
@@ -341,6 +470,30 @@ func (m *Module) GetMap(mapName string) (*BPFMap, error) {
...
+
+func (b *BPFMap) Type() MapType {
+	return MapType(C.bpf_map__type(b.bpfMap))
+}
+
+// SetType is used to set the type of a bpf map that isn't associated
+// with a file descriptor already. If the map is already associated
+// with a file descriptor the libbpf API will return error code EBUSY
+func (b *BPFMap) SetType(mapType MapType) error {
+	errC := C.bpf_map__set_type(b.bpfMap, C.enum_bpf_map_type(int(mapType)))
```
Comment (yanivagman) [inline libbpfgo.go:490]: same as above

**Review by yanivagman — APPROVED** [review]
Nice addition @grantseltzer 
LGTM, just a small comment/question

**Pull request merged**